### PR TITLE
GCW-3583 System Messages

### PR DESCRIPTION
**Pull request by bharat619**
### Ticket Link: 
https://jira.crossroads.org.hk/browse/GCW-3538

### What does this PR do?
Make system message for "new offer submitted" editable

### CODE REVIEW

File: app/controllers/api/v1/canned_responses_controller.rb [hunk 1]
```diff
@@ -4,28 +4,60 @@ module Api
...
+          param :name_zh_tw, String, desc: 'Chinese name for the canned_response', required: false, allow_nil: true
+          param :content_en, String, desc: 'English content for the canned_response', required: true
+          param :content_zh_tw, String, desc: 'Chinese content for the canned_response', required: false, allow_nil: true
+        end
+      end
+
+      api :GET, '/v1/canned_responses', 'List all private / non private canned_responses'
+      param :isPrivate, String, desc: 'Flag to indicate private / non private message'
       def index
+        is_private = bool_cast(params['isPrivate']) || false
```
Comment (abulsayyad123) [inline app/controllers/api/v1/canned_responses_controller.rb:22]: You can use `bool_param` here it will return a boolean value.

**Review by abulsayyad123 — APPROVED** [review]

File: app/controllers/api/v1/canned_responses_controller.rb [hunk 2]
```diff
@@ -4,28 +4,60 @@ module Api
   module V1
     # Api::V1::CannedResponseController
     class CannedResponsesController < Api::V1::ApiController
-      load_and_authorize_resource :canned_response, parent: false
+      load_and_authorize_resource :canned_response, parent: false, except: :show
+      skip_authorization_check only: :show
```
Comment (steveyken) [inline app/controllers/api/v1/canned_responses_controller.rb:8]: I think we want `show` available to only admins?

File: app/models/canned_response.rb [hunk 3]
```diff
@@ -14,4 +14,15 @@ class CannedResponse < ApplicationRecord
...
   validates_presence_of :content_en
+  validates :guid, uniqueness: true, allow_nil: true
+
+  before_destroy :validate_for_private_messages
+
+  scope :by_private, ->(is_private) { where(is_private: is_private) }
+
+  private
+
+  def validate_for_private_messages
```
Comment (steveyken) [inline app/models/canned_response.rb:25]: Can you explain what this is for?

**Review by steveyken — CHANGES_REQUESTED** [review]
Thanks @bharat619 apologies this took a while to review. Couple of questions below but think we can clear them up pretty quick. thanks

File: app/controllers/api/v1/canned_responses_controller.rb (hunk 2 above)
Comment (bharat619) [reply app/controllers/api/v1/canned_responses_controller.rb:8]: sure, ill change it

File: app/models/canned_response.rb (hunk 3 above)
Comment (bharat619) [reply app/models/canned_response.rb:25]: A validation to ensure no deletion happens for private message

`before_destroy: validate_for_private_messages`

File: app/controllers/api/v1/canned_responses_controller.rb [hunk 4]
```diff
@@ -6,26 +6,57 @@ module V1
...
+        param :canned_response, Hash, required: true do
+          param :name_en, String, desc: 'English name for the canned_response', required: true
+          param :name_zh_tw, String, desc: 'Chinese name for the canned_response', required: false, allow_nil: true
+          param :content_en, String, desc: 'English content for the canned_response', required: true
+          param :content_zh_tw, String, desc: 'Chinese content for the canned_response', required: false, allow_nil: true
+        end
+      end
+
+      api :GET, '/v1/canned_responses', 'List all private / non private canned_responses'
+      param :isPrivate, String, desc: 'Flag to indicate private / non private message'
```
Comment (steveyken) [inline app/controllers/api/v1/canned_responses_controller.rb:19]: @bharat619 can you update isPrivate to preventDelete everywhere?

**Review by steveyken — APPROVED** [review]
This is good to go. Well done

**Pull request merged**